widgets/btc_linux.lua

```diff
@@ -0,0 +1,54 @@
+---------------------------------------------------
+-- Licensed under the GNU General Public License v2
+--  * (c) 2017, 0x5b <dragen15051@gmail.com>
+---------------------------------------------------
+
+-- {{{ Grab environment
+local setmetatable = setmetatable
+local helpers = require("vicious.helpers")
+local https = require("socket.http")
+local json = require("json")
+local string = {
+    sub = string.sub,
+    upper = string.upper,
+}
+-- }}}
+
+
+-- Btc: provides current bitcoin price
+-- vicious.widgets.btc
+local btc_linux = {}
+
+
+-- {{ Bitcoin widget type
+local function worker(format, warg)
+
+    -- Default values
+    local btc = { ["{price}"] = "N/A" }
+    local response, code = nil, nil
+
+    local currency_code = string.upper(warg)
+    local url = "https://api.coindesk.com/v1/bpi/currentprice/"..helpers.escape(currency_code)..".json"
+
+    -- Get current price
+    response, code = https.request(url)
```

Comment:
Wow do you avoid blocking here, when network is not reachable?

Reply:
I dont know lua well, but testing in lua repl showed: if network is not reachable, ```https.request()``` returns ```nil``` instantly. I could be wrong...

Reply:
well, in weather widget, information was received by ```curl```, i can do the same

Reply:
the weather widget is also buggy in this regard as it will also block. You can look at `pkg_all.lua` to see how commands can be called asynchronous.

Reply:
@Mic92 , done, looks like async call..